File: Polmorfismo/Ejer3/Restaurante.py
class Cocinero:
    def __init__(self, nombre, sueldo_mensual, horas_extras, sueldo_por_hora):
        self.nombre = nombre
        self.sueldo_mensual = sueldo_mensual
        self.horas_extras = horas_extras
        self.sueldo_por_hora = sueldo_por_hora

    def calcular_sueldo_total(self):
        sueldo_total = self.sueldo_mensual + (self.horas_extras * self.sueldo_por_hora)
        print(f"{self.nombre} es Cocinero y su sueldo total es: {sueldo_total}")

    def get_sueldo_mensual(self):
        return self.sueldo_mensual

    def mostrar_informacion(self):
        print(f"Nombre: {self.nombre}, Sueldo: {self.sueldo_mensual}")


class Mesero:
    def __init__(self, nombre, sueldo_mensual, horas_extras, sueldo_por_hora, propinas):
        self.nombre = nombre
        self.sueldo_mensual = sueldo_mensual
        self.horas_extras = horas_extras
        self.sueldo_por_hora = sueldo_por_hora
        self.propinas = propinas

    def calcular_sueldo_total(self):
        sueldo_total = self.sueldo_mensual + (self.horas_extras * self.sueldo_por_hora) + self.propinas
        print(f"{self.nombre} es Mesero y su sueldo total es: {sueldo_total}")

    def get_sueldo_mensual(self):
        return self.sueldo_mensual

    def mostrar_informacion(self):
        print(f"Nombre: {self.nombre}, Sueldo: {self.sueldo_mensual}")


class Administrativo:
    def __init__(self, nombre, sueldo_mensual, cargo):
        self.nombre = nombre
        self.sueldo_mensual = sueldo_mensual
        self.cargo = cargo

    def calcular_sueldo_total(self):
        print(f"{self.nombre} es Administrativo y su sueldo total es: {self.sueldo_mensual}")

    def get_sueldo_mensual(self):
        return self.sueldo_mensual

    def mostrar_informacion(self):
        print(f"Nombre: {self.nombre}, Sueldo: {self.sueldo_mensual}, Cargo: {self.cargo}")


if __name__ == "__main__":
    cocinero1 = Cocinero("Carlos", 2500, 10, 15)
    mesero1 = Mesero("Luis", 1800, 5, 12, 200)
    mesero2 = Mesero("Ana", 1700, 8, 12, 150)
    administrativo1 = Administrativo("Marta", 3000, "Gerente")
    administrativo2 = Administrativo("Pedro", 2500, "Supervisor")

    print("----------------Calculo de sueldos totales--------------")
    cocinero1.calcular_sueldo_total()
    mesero1.calcular_sueldo_total()
    mesero2.calcular_sueldo_total()
    administrativo1.calcular_sueldo_total()
    administrativo2.calcular_sueldo_total()

    sueldo_buscado = 2500
    print("--------------------------------------------------------")
    print(f"Empleados con sueldo igual a {sueldo_buscado}:")
    if cocinero1.get_sueldo_mensual() == sueldo_buscado:
        cocinero1.mostrar_informacion()
    if mesero1.get_sueldo_mensual() == sueldo_buscado:
        mesero1.mostrar_informacion()
    if mesero2.get_sueldo_mensual() == sueldo_buscado:
        mesero2.mostrar_informacion()
    if administrativo1.get_sueldo_mensual() == sueldo_buscado:
        administrativo1.mostrar_informacion()
    if administrativo2.get_sueldo_mensual() == sueldo_buscado:
        administrativo2.mostrar_informacion()

File: Polmorfismo/Ejer3/Restaurante.java
package Polmorfismo.Ejer3;

public class Restaurante {
    public static void main(String[] args) {
        Cocinero cocinero1 = new Cocinero("Carlos", 2500, 10, 15);
        Mesero mesero1 = new Mesero("Luis", 1800, 5, 12, 200);
        Mesero mesero2 = new Mesero("Ana", 1700, 8, 12, 150);
        Administrativo administrativo1 = new Administrativo("Marta", 3000, "Gerente");
        Administrativo administrativo2 = new Administrativo("Pedro", 2500, "Supervisor");

        System.out.println("----------------Calculo de sueldos totales--------------");
        cocinero1.calcularSueldoTotal();
        mesero1.calcularSueldoTotal();
        mesero2.calcularSueldoTotal();
        administrativo1.calcularSueldoTotal();
        administrativo2.calcularSueldoTotal();

        float sueldoBuscado = 2500;
        System.out.println("--------------------------------------------------------");
        System.out.println("Empleados con sueldo igual a " + sueldoBuscado + ":");
        if (cocinero1.getSueldoMensual() == sueldoBuscado) cocinero1.mostrarInformacion();
        if (mesero1.getSueldoMensual() == sueldoBuscado) mesero1.mostrarInformacion();
        if (mesero2.getSueldoMensual() == sueldoBuscado) mesero2.mostrarInformacion();
        if (administrativo1.getSueldoMensual() == sueldoBuscado) administrativo1.mostrarInformacion();
        if (administrativo2.getSueldoMensual() == sueldoBuscado) administrativo2.mostrarInformacion();
    }
}

class Cocinero {
    private String nombre;
    private float sueldoMensual, sueldoPorHora;
    private int horasExtras;

    public Cocinero(String nombre, float sueldoMensual, int horasExtras, float sueldoPorHora) {
        this.nombre = nombre;
        this.sueldoMensual = sueldoMensual;
        this.horasExtras = horasExtras;
        this.sueldoPorHora = sueldoPorHora;
    }

    public void calcularSueldoTotal() {
        float sueldoTotal = sueldoMensual + (horasExtras * sueldoPorHora);
        System.out.println(nombre + " es Cocinero y su sueldo total es: " + sueldoTotal);
    }

    public float getSueldoMensual() {
        return sueldoMensual;
    }

    public void mostrarInformacion() {
        System.out.println("Nombre: " + nombre + ", Sueldo: " + sueldoMensual);
    }
}

class Mesero {
    private String nombre;
    private float sueldoMensual, sueldoPorHora, propinas;
    private int horasExtras;

    public Mesero(String nombre, float sueldoMensual, int horasExtras, float sueldoPorHora, float propinas) {
        this.nombre = nombre;
        this.sueldoMensual = sueldoMensual;
        this.horasExtras = horasExtras;
        this.sueldoPorHora = sueldoPorHora;
        this.propinas = propinas;
    }

    public void calcularSueldoTotal() {
        float sueldoTotal = sueldoMensual + (horasExtras * sueldoPorHora) + propinas;
        System.out.println(nombre + " es Mesero y su sueldo total es: " + sueldoTotal);
    }

    public float getSueldoMensual() {
        return sueldoMensual;
    }

    public void mostrarInformacion() {
        System.out.println("Nombre: " + nombre + ", Sueldo: " + sueldoMensual);
    }
}

class Administrativo {
    private String nombre, cargo;
    private float sueldoMensual;

    public Administrativo(String nombre, float sueldoMensual, String cargo) {
        this.nombre = nombre;
        this.sueldoMensual = sueldoMensual;
        this.cargo = cargo;
    }

    public void calcularSueldoTotal() {
        System.out.println(nombre + " es Administrativo y su sueldo total es: " + sueldoMensual);
    }

    public float getSueldoMensual() {
        return sueldoMensual;
    }

    public void mostrarInformacion() {
        System.out.println("Nombre: " + nombre + ", Sueldo: " + sueldoMensual + ", Cargo: " + cargo);
    }
}
